src/main/java/placement/model/board/Board.java

```diff
@@ -0,0 +1,180 @@
+    private final int cols;
+
+    /**
+     * creates a Board object with preset dimensions.
+     */
+    private Board(int rows, int cols) {
+        boardFields = new BoardFields(rows, cols);
+        this.rows = rows;
+        this.cols = cols;
+    }
+
+    /**
+     * Factory method for creating 10x10 board.
+     *
+     * @return Board of size 10x10
+     */
+    public static Board createDefaultBoard() {
+        return new Board(DEFAULT_ROW_COUNT, DEFAULT_COLUMN_COUNT);
+    }
+
+    /**
+     * Returns true if a board is not yet populated.
+     *
+     * @return true if a board is not yet populated.
+     */
+    boolean isEmpty() {
+        return boardFields.isEmpty();
+    }
+
+    /**
+     * Returns number of rows in the board (y-size).
+     *
+     * @return number of rows in the board.
+     */
+    public int rows() {
+        return rows;
+    }
+
+    /**
+     * Returns number of columns in the board (x-size).
+     *
+     * @return number of columns in the board.
+     */
+    public int cols() {
+        return cols;
+    }
+
+
+    private FieldState getFieldState(Coordinates coordinates) {
+        Field field = boardFields.getField(coordinates);
+        return field.getState();
+    }
+
+    FieldState getFieldState(int x, int y) {
+        Coordinates coordinates = new Coordinates(x, y);
+        return getFieldState(coordinates);
+    }
+
+    /**
+     * Places given ship on the board if it's possible.
+     *
+     * @param ship        ship to place.
+     * @param coordinates coordinates of the ship's "head".
+     * @see ShipOnBufferException
+     * @see ShipOnOccupiedFieldException
+     * @see ShipOutOfBoardException
+     */
+    public void placeShip(Ship ship, Coordinates coordinates) {
+        try {
+            Set<Coordinates> mastCoordinates = getMastCoordinates(ship, coordinates);
+            mastCoordinates.forEach(coordinate -> {
+                boardFields.markAsOccupied(coordinate);
+                surroundWithBuffer(coordinate);
+                ship.addPositionCoord(coordinate);
+            });
+            ship.markAsPlaced();
+
+        } catch (ShipOutOfBoardException | ShipOnBufferException | ShipOnOccupiedFieldException e) {
+            System.err.println(e.getMessage());
+        }
+    }
+
+    public void placeShip(Ship ship, int x, int y) {
+        Coordinates coordinates = new Coordinates(x, y);
+        placeShip(ship, coordinates);
+    }
+
+    private Set<Coordinates> getMastCoordinates(Ship ship, Coordinates coordinates)
+            throws ShipOutOfBoardException, ShipOnBufferException, ShipOnOccupiedFieldException {
+        Set<Coordinates> mastCoordinates = new HashSet<>();
+        while (mastCoordinates.size() < ship.getMastNumber()) {
+            checkIsAbleToPlace(coordinates);
+            Coordinates nextCoord = ship.nextCoordinates();
+            mastCoordinates.add(coordinates);
+            coordinates = coordinates.add(nextCoord);
```

Comment:
it looks suspicious.
Why do you want to change a reference to the method argument? 
Why do you add coordinates?